### PR TITLE
Require the competitor to finish signing

### LINKED ISSUE #859: Require the competitor to finish signing (not just for the judge to finish signing) before the competitor may reset the timer.
I'm not sure we have an issue for this yet, so I'm filing a separate one.

### PR DESCRIPTION
**Pull request by Ivan-Brigidano**
Closes #859. Thanks @Goosly for spotting it!
I'm not happy with the wording, can someone please help me out?

### CODE REVIEW

File: wca-regulations.md
Comment (Claster) [inline wca-regulations.md]: A7c4 mentions "the judge has recorded and signed the attempt". So maybe replace "has signed the result on the score sheet" to "has signed the attempt" for consistency?

**Review by Claster — COMMENTED** [review]
LGTM modulo this comment

File: wca-regulations.md
Comment (Ivan-Brigidano) [reply wca-regulations.md]:
```suggestion
    - A6f) The competitor must not reset the timer until both the competitor and the judge has signed the attempt. Penalty: disqualification of the attempt (DNF), at the discretion of the judge.
```

**Comment by xsrvmy** [comment]
I still have a legilimate concern woth the change itself: It is much more likely that this reg gets broken on a solve with a lengthy reaction (usually personal records).

**Review by Rouxles — CHANGES_REQUESTED** [review]
> The competitor must not reset the timer until both the competitor and the judge **has** signed the attempt. Penalty: disqualification of the attempt (DNF), at the discretion of the judge.

I think "has" here should be replaced with "have"

**Comment by hanwu85** [comment]
@xsrvmy , but after the long reaction, the competitor still has to comeback and sign the score sheet.
I think the competitor can not reset the timer no matter what; it is the competitor who has to pay attention.
Or are you referring to lengthy celebration and forgot to sign?

File: wca-regulations.md
Comment (Ivan-Brigidano) [inline wca-regulations.md]:
```suggestion
    - A6f) The competitor must not reset the timer until both the competitor and the judge have signed the attempt. Penalty: disqualification of the attempt (DNF), at the discretion of the judge.
```

**Comment by xsrvmy** [comment]
> @xsrvmy , but after the long reaction, the competitor still has to comeback and sign the score sheet.
> I think the competitor can not reset the timer no matter what; it is the competitor who has to pay attention.
> Or are you referring to lengthy celebration and forgot to sign?

Several possibilities:
1. judge resets timer by mistake
2. competitor loses the flow of the procedure and resets the timer by mistake
3. judge takes scoresheet without competitor signing

Actually now that I think about it, what would happen if the judge forgot to let the competitor sign, the competitor does not realize, and realizes 10 minutes later?

**Comment by SAuroux** [comment]
Thinking a bit more about this, this is severely broken. To apply A7c4+ now, the competitor would need to proof that he wasn't the person that has reset the timer (which is pretty much impossible).

**Comment by xsrvmy** [comment]
Yeah I think that's the biggest objection from the community @SAuroux 
Given the problem this is trying to solve I think such a change is overkill.
Question for the WRC actually: The reason given on the list of proposed changes was to resolve incidents after the timer has been stopped, but I cannot think of a specific example that this change would fix.

**Comment by Ivan-Brigidano** [comment]
> Actually now that I think about it, what would happen if the judge forgot to let the competitor sign, the competitor does not realize, and realizes 10 minutes later?

Delegate might use the [guideline A7c4+](https://www.worldcubeassociation.org/regulations/guidelines.html#A7c4+)

> Thinking a bit more about this, this is severely broken. To apply A7c4++ now, the competitor would need to proof that he wasn't the person that has reset the timer (which is pretty much impossible).

@SAuroux I'm not understanding you properly. There is no guideline A7c4++, I suppose you are referring to A7c4+. I can't seed any problem to continue applying this guideline, maybe I'm missing something important (?).

> Yeah I think that's the biggest objection from the community @SAuroux
> Given the problem this is trying to solve I think such a change is overkill.
> Question for the WRC actually: The reason given on the list of proposed changes was to resolve incidents after the timer has been stopped, but I cannot think of a specific example that this change would fix.

I can't see any objections on the issue and didn't receive any mail complaining about it. So, apparently, there is no big community objection.
I have seen several cases where the competitor rests the timer before signing. And most of the times they didn't reviewed if the judge wrote the score properly. I think this introduces consistency and granted a cleaner process of signing.

**Comment by xsrvmy** [comment]
@Ivan-Brigidano I think the argument is whether or not A7c4+ supersedes this change. If it does not, then A7c4+ becomes pointless. If it does, then a competitor can claim he forgot to sign and get away with it.

**Comment by Goosly** [comment]
> I can't seed any problem to continue applying this guideline, maybe I'm missing something important (?).

If we find a scorecard with a competitor signature missing, how do we know he did not reset the timer (which would violate the new A6f)?
Does A7c4+ overrule A6f or does A6f overrule A7c4+?

**Comment by xsrvmy** [comment]
Actuallt, now I think about it, this was a problem before if the judge forgot to sign as well.
Also there is an incident on the log where a solve with missing signatures was accepted because judge forgot to let anyone sign at all, but that was 5bld probably with a stopwatch.

**Comment by Goosly** [comment]
> Actuallt, now I think about it, this was a problem before if the judge forgot to sign as well.

No, this was not a problem, since the current regulations say:

> The competitor must not reset the timer until the judge has recorded the result on the score sheet.

And when we apply A7c4+, the result is recorded. It's only missing signatures.

**Comment by Ivan-Brigidano** [comment]
@SAuroux @Goosly Delegates would still be able to use A7c4+, do you have any thoughts to clarify it?

**Comment by Laura-O** [comment]
@Ivan-Brigidano I think in order to resolve this properly it would be helpful if you would reply to the questions. @Goosly asked:

> Does A7c4+ overrule A6f or does A6f overrule A7c4+?

Let me also rephrase this question:
If I find a missing competitor signature, I usually try to resolve this by finding the competitor and letting him/her sign. If this change gets implemented, I would have to check if the timer was reset. If this is the case, would I have to ask the judge if this should be a DNF or can I decide to let the competitor sign (and overrule the judge)?

**Comment by Ivan-Brigidano** [comment]
> @Ivan-Brigidano I think in order to resolve this properly it would be helpful if you would reply to the questions. @Goosly asked:
> 
> > Does A7c4+ overrule A6f or does A6f overrule A7c4+?
> 
> Let me also rephrase this question:
> If I find a missing competitor signature, I usually try to resolve this by finding the competitor and letting him/her sign. If this change gets implemented, I would have to check if the timer was reset. If this is the case, would I have to ask the judge if this should be a DNF or can I decide to let the competitor sign (and overrule the judge)?

I think the most reasonable way is to overrule the judge at the Delegate discretion. So yes, (in my opinion) A7c4+ overrule A6f.
How do you think we can clarify how to act in these situations?

**Comment by Ivan-Brigidano** [comment]
@thewca/wrc-team any thoughts?

**Comment by lgarron** [comment]
> @thewca/wrc-team any thoughts?

I've thought about this a bit. While I think we could find a way to get this right, I don't see a great danger in leaving the change out of the 2020 Regs. So I think we should not land it for now.

**Comment by Ivan-Brigidano** [comment]
> > @thewca/wrc-team any thoughts?
> 
> I've thought about this a bit. While I think we could find a way to get this right, I don't see a great danger in leaving the change out of the 2020 Regs. So I think we should not land it for now.

Then, I'll close the PR

**Comment by xsrvmy** [comment]
Should someone create an issue to discuss this? I don't think there was one.

**Comment by Ivan-Brigidano** [comment]
> Should someone create an issue to discuss this? I don't think there was one.

#859 you indeed posted come comments there

**Pull request closed without merging**
